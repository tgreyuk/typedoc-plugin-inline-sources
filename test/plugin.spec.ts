import { Application, ProjectReflection, TSConfigReader } from 'typedoc';

describe(`Plugin:`, () => {
  let app: Application;
  let project: ProjectReflection;

  beforeAll(() => {
    app = new Application();
    app.options.addReader(new TSConfigReader());
    app.bootstrap({
      tsconfig: '../tsconfig.test.json',
    });
    project = app.convert() as ProjectReflection;
  });

  test(`should inject variable source`, async () => {
    const reflection = project.findReflectionByName('message');
    expect(reflection?.comment?.tags).toMatchSnapshot();
  });

  test(`should inject type alias source`, async () => {
    const reflection = project.findReflectionByName('ID');
    expect(reflection?.comment?.tags).toMatchSnapshot();
  });

  test(`should inject function source`, async () => {
    const reflection = project.findReflectionByName('doSomething') as any;
    expect(reflection.signatures[0].comment?.tags).toMatchSnapshot();
  });

  test(`should inject class source`, async () => {
    const reflection = project.findReflectionByName('GoodGreeter') as any;
    expect(reflection.comment?.tags).toMatchSnapshot();
  });

  test(`should inject class method source`, async () => {
    const reflection = project
      ?.findReflectionByName('Rectangle')
      ?.findReflectionByName('calcArea') as any;
    expect(reflection.signatures[0].comment?.tags).toMatchSnapshot();
  });

  test(`should inject class accessor source`, async () => {
    const reflection = project
      ?.findReflectionByName('Rectangle')
      ?.findReflectionByName('area') as any;
    expect(reflection.comment?.tags).toMatchSnapshot();
  });

  test(`should inject interface source`, async () => {
    const reflection = project?.findReflectionByName('Person') as any;
    expect(reflection.comment?.tags).toMatchSnapshot();
  });
});
