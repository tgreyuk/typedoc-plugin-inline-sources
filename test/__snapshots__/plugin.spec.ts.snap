// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin: should inject class accessor source 1`] = `
Array [
  Object {
    "paramName": "",
    "tagName": "source",
    "text": "Class getter source

\`\`\`typescript
get area() {
    return this.calcArea();
}
\`\`\`

",
  },
]
`;

exports[`Plugin: should inject class method source 1`] = `
Array [
  Object {
    "paramName": "",
    "tagName": "source",
    "text": "Class method source

\`\`\`typescript
calcArea() {
    return this.height * this.width;
}
\`\`\`

",
  },
]
`;

exports[`Plugin: should inject class source 1`] = `
Array [
  Object {
    "paramName": "",
    "tagName": "source",
    "text": "Class source

\`\`\`typescript
export class GoodGreeter {
    name: string;
    constructor() {
        this.name = 'hello';
    }
}
\`\`\`

",
  },
]
`;

exports[`Plugin: should inject function source 1`] = `
Array [
  Object {
    "paramName": "",
    "tagName": "source",
    "text": "Function source

\`\`\`typescript
export function doSomething(x: string | null) {
    if (x === null) {
        /* do nothing */
    }
    else {
        // do something
        console.log('Hello, ' + x.toUpperCase());
    }
}
\`\`\`

",
  },
]
`;

exports[`Plugin: should inject interface source 1`] = `
Array [
  Object {
    "paramName": "",
    "tagName": "source",
    "text": "Interface source

\`\`\`typescript
export interface Person {
    name: string;
    age: number;
}
\`\`\`

",
  },
]
`;

exports[`Plugin: should inject type alias source 1`] = `
Array [
  Object {
    "paramName": "",
    "tagName": "source",
    "text": "Type Alias source

\`\`\`typescript
export type ID = number | string;
\`\`\`

",
  },
]
`;

exports[`Plugin: should inject variable source 1`] = `
Array [
  Object {
    "paramName": "",
    "tagName": "source",
    "text": "source tag comment?

\`\`\`typescript
const message = 'Hello World!'
\`\`\`

",
  },
]
`;
